### ISSUE TITLE
classification2

### ISSUE THREAD

**Issue by GloriaWYY** [UBC-DSCI/introduction-to-datascience-python#9]
Branch: [classification2](https://github.com/UBC-DSCI/introduction-to-datascience-python/tree/classification2)
PR opened: PR #10 , the [deployed Chapter](https://gloriawyy.github.io/classification2/classification2.html)

**Comment by GloriaWYY** [comment]
## General progress:

- Around 80% of the R code presented to students has been converted into Python.
- I still need to convert the remaining code and adapt the text description based on Python equivalence.

## Some questions:

1. In worksheet, for the function `cross_validation`, it asks the students to set `return_train_score=True` to return the training score in addition to the validation score. But in R textbook, this is not mentioned, and the R function `collect_metrics` does not return training metrics. Should I introduce `return_train_score` in the Python textbook?

    Yes.

2. In the section [`6.6.2 Parameter value selection`](https://datasciencebook.ca/classification2.html#parameter-value-selection), the R textbook introduces `tune_grid()` to tune the classifier. In `scikit-learn`, `GridSearchCV` and `RandomizedSearchCV` are used to tune hyperparameters, and these are both introduced in [worksheet](https://github.ubc.ca/UBC-DSCI/dsci-100-instructor/blame/py_worksheet_classification2/source/py_worksheet_classification2/py_worksheet_classification2.ipynb#L1579). Should I introduce both of them? (This specific section in textbook is using grid search.)

    Yes, can copy/adapt what is already in Python worksheet.

3. In the section [`6.8.3 Forward selection in R`](https://datasciencebook.ca/classification2.html#forward-selection-in-r), the text explains the for loop and refers to an R book (R for Data Science). Should I find for loop documentation/book resources in Python and replace the R resource?

    Yes, here are some resources that might be useful:

    a. https://wesmckinney.com/book/python-basics.html#control_for
    b. https://swcarpentry.github.io/python-novice-inflammation/05-loop/index.html
    c. https://jakevdp.github.io/PythonDataScienceHandbook/
    d. https://phaustin.github.io/Problem-Solving-with-Python/Loops/For-Loops.html
    e. https://www.bigbookofpython.com/programming.html
    

4. In the section [`6.8.3 Forward selection in R`](https://datasciencebook.ca/classification2.html#forward-selection-in-r), the R textbook presents the code to show how each predictor is selected over iterations in forward selection, and mentions that "there is no built-in way to do this using the `tidymodels` framework". In Python, `SequentialFeatureSelector` can automate the forward selection as a step in the pipeline, but to my knowledge it cannot give accuracy scores during each iteration. So should I introduce  `SequentialFeatureSelector` aside from coding the demonstration similar to R textbook?

    Yes, introduce `SequentialFeatureSelector` in the text, and also write the code to show the similar outputs (*i.e.* validation accuracy over iterations) as in R textbook.
